connection.go

```diff
@@ -25,10 +27,16 @@ type fireboltConnection struct {
 // Prepare returns a firebolt prepared statement
 // returns an error if the connection isn't initialized or closed
 func (c *fireboltConnection) Prepare(query string) (driver.Stmt, error) {
+	return c.PrepareContext(context.TODO(), query)
+}
+
+// PrepareContext returns a firebolt prepared statement
+// returns an error if the connection isn't initialized or closed
+func (c *fireboltConnection) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
 	if c.client != nil && len(c.engineUrl) != 0 {
-		return &fireboltStmt{execer: c, queryer: c, query: query}, nil
+		return statement.MakeStmt(c, query, contextUtils.GetPreparedStatementsStyle(ctx))
 	}
-	return nil, errors.New("fireboltConnection isn't properly initialized")
+	return nil, errors.New("firebolt connection isn't properly initialized")
```

Comment:
Seems a bit redundant to me to make from one method two where the purpose of one is just to call the other...

Reply:
This is done in order to implement the library interface. These both methods are present in go sql